src/Avalonia.Base/Data/Core/SettableNode.cs

```diff
@@ -15,7 +15,8 @@ public bool SetTargetValue(object value, BindingPriority priority)
 
         private bool ShouldNotSet(object value)
         {
-            if (PropertyType == null)
+            var propertyType = PropertyType;
```

Comment:
PropertyType can use reflection internally, so it makes sense to reuse its value.